### ISSUE TITLE
Should we remove ELF games from training?

### ISSUE THREAD

**Issue by Friday9i** [gcp/leela-zero#2011]
LZ is at least as strong as Elf since we switched to 40 blocks, ie LZ174.
However, despite significant improvements (15 new nets, with LZ189 now), performance of LZ against Elf is still relatively weak ("only" 59% from LZ187). 
And, surprisingly, LZ doesn't scale well against the smaller Elf net with more visits: see the scaling curves here #1914. The only decreasing curve of a better net vs a weaker net is LZ181 vs ELF, but LZ188 unfortunately shows the same deceiving result (I'm finishing the picture of LZ188 vs ELF probably tonight, with significant help from @herazul to draw it).
What is surprising is the bad scaling: generally, larger nets (and better-trained nets) show an increased ratio with more visits (above ~100 visits), ie the curve goes up and has a U-shape. But for the 40b net vs the 20b Elf net, it decreases with more visits, which means ELF is still significantly better than LZ188 for long and deep games ; -(.

So, here is my understanding of the situation:
**Theory:** trained with ELF games, LZ logically learns to mimick it.
**Hypothesis:** mimicking ELF does not allow LZ to become significantly better than it! That seems perfectly reasonable: if LZ analyses the same moves as ELF, with comparable evaluations, it cannot play significantly better despite having more potential from 40b. Worse: on time parity, it will be crushed by ELF as it goes less deep than ELF (which runs faster). Hence, successive LZ nets can improve against themselves, one after the other, and against LZ174, but they cannot really improve against ELF (as LZ is still mimicking ELF). Hence, if true, training from ELF is now a problem (vs an advantage before)...
**Testing the hypothesis:** removing ELF games from the training pipeline will give us the answer. If the hypothesis is true, LZ nets:
- should still improve as they are today (and probably even faster, with promotions more frequently)
- but they should also begin to scale much better against ELF ;-))): the scalability curve should become an U-curve (LZ should perfom better against ELF with more visits) and the winrate in matches against ELF should go up much faster!

**Conclusion:** now that LZ is at least as strong as ELF, we should probably remove all ELF games from the training pipeline! (Moreover, ELF games are "static", which could lead to overfitting).
**Good news:** @gcp intends to do it "soon", from what he shared on another thread, so we'll get the answer soon ;-).

_Note:_ previous tests to train nets without ELF games were done, to my knowledge, with 20b nets (or very early 40b?). Results were poor as trained nets were weaker than nets trained with ELF... But that was at a time where ELF was better than LZ, so the situation is not comparable to the situation today, where LZ is significanty better than ELF.

**Comment by l1t1** [comment]
@bjiyxo may be know

**Comment by wonderingabout** [comment]
@Friday9i 
leela zero is not weak, ELF may be 20b, but it's just ELF that is very strong and probably stronger than leela zero especially on middle game
if leela zero has higher winrate overall in 400 games, it's just because it understand the early game enough to not be left behind and often takes a small or sometimes significant lead
there's also the issue of elf having sharp winrates, therefore resigning statistically more, when leela zero will "tryhard" and sometimes win

personally, i'm looking forward to facebook's next release of elf
i dont doubt it will be very strong and interesting to see if it is a 40b net

from my and other people's testing, elf is certainly not weaker than leela zero in the middle game at evaluating any board position, and leela zero as it is cant be able to fully replace it for reviews and insights

**Comment by john45678** [comment]
@wonderingabout  How do you know FB is going to upgrade Elf again?

**Comment by jillybob** [comment]
The recent increase in strength was due to removing ELF games, so SLOWLY removing more ELF games should result in more strength increase.

**Comment by l1t1** [comment]
i notice some code added  at elf's site(https://github.com/pytorch/ELF/tree/newtasks), and  got no answer

https://github.com/pytorch/ELF/issues/103

**Comment by alphaladder** [comment]
Wishful thinking of some guys.

**Comment by gcp** [comment]
ELF games have been totally removed a few days ago, I think 6c7 was the first network to see the effect.

**Comment by l1t1** [comment]
6c7 promoted just after leelaz 0.16
